src/COGPaper/heuristics/StateHeuristic.java

```diff
@@ -33,21 +41,60 @@
 
     public void initHeuristicAccumulation() {
         heuristic_acc = 0;
+        heuristic_acc_counter = 0;
+        extreme_positive = false;
+        extreme_negative = false;
         last_stateObs = last_visited_stateObs;
     }
 
     public void accumulateHeuristic(StateObservation stateObs) {
-        heuristic_acc += evaluateState(stateObs);
+        double heuristic_value = evaluateState(stateObs);
```

Comment:
I think a NaN check would be in order here.  If evaluateState() returns NaN, then it is possible the Double.compare() calls will not do what we want.

Reply:
You are right, I have included it; thanks!